Neos.Fusion/Classes/Core/Cache/ParserCacheIdentifierTrait.php

```diff
@@ -19,26 +19,24 @@
 trait ParserCacheIdentifierTrait
 {
     /**
-     * creates a comparable hash of the dsl type and content to be used as cache identifier
+     * Creates a comparable hash of the dsl type and content to be used as cache identifier
      */
     private function getCacheIdentifierForDslCode(string $identifier, string $code): string
     {
         return 'dsl_' . $identifier . '_' . md5($code);
     }
 
     /**
-     * creates a comparable hash of the absolute, resolved $fusionFileName
+     * Creates a comparable hash of the absolute $absoluteFusionFilePath
+     *
+     * it is crucial that the $absoluteFusionFilePath doesn't contain directory traversal
+     * like /../ as this will result in a different hash and cannot be flushed!
```

Comment:
I'm pretty sure we shoud guarantee this in the method, like before. Check the `realptah()` result, and ignore it. Optionally check for traversal parts and throw only, when `realpath()` fails **and** there is a traversal part?

Don't we have something to normalize paths in our uitility classes?

Reply:
The files utility doenst have any helping hand regarding directory traversals. (except to rewrite them to unix format)

https://github.com/neos/flow-development-collection/pull/3235 shows a new `Files::isRealPath` which can be used to check if a path is absolute and has no directory traversal.

That check might be a good assertion, so if flow changes its api https://github.com/neos/neos-development-collection/pull/4838#discussion_r1453472514 we notice that.

Reply:
luckily $pathFromFileMonitor its already in unix style format:

https://github.com/neos/flow-development-collection/blob/dad697040135e49a9efec11e6973ceecf27bfb50/Neos.Flow/Classes/Monitor/FileMonitor.php#L216

so yes this assumption is correct.